source/common/common/macros.h

```diff
@@ -16,4 +16,15 @@ namespace Envoy {
  * Stop the compiler from complaining about an unreferenced parameter.
  */
 #define UNREFERENCED_PARAMETER(X) ((void)(X))
+
+/**
+ * Have a generic fall-through for different versions of C++
+ */
+#if __cplusplus == 201103L || __cplusplus == 201402L // C++11
+#define FALLTHRU 
+#elif __cplusplus > 201402L // C++14 and C++17 and above
+#define FALLTHRU [[fallthrough]]
+#else // C and C++03
+#define FALLTHRU __attribute__ ((fallthrough))
```

Comment:
Fair enough, maybe just add a comment or link to some reference if there is one.